### PR TITLE
[enhancement]optimize some bitmap function logic

### PR DESCRIPTION
**Pull request by ucasfl**
# Proposed changes

Issue Number: close #xxx

Optimize logic of bitmap function `bitmap_subset_in_range` and `bitmap_subset_limit`: pass arguments by value for basic type, and reduce branch stantement for better vectorization.

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/util/bitmap_value.h [hunk 1]
```diff
@@ -1695,8 +1695,7 @@ class BitmapValue {
     /**
      * Return new set with specified range (not include the range_end)
      */
-    int64_t sub_range(const int64_t& range_start, const int64_t& range_end,
-                      BitmapValue* ret_bitmap) {
+    int64_t sub_range(int64_t range_start, int64_t range_end, BitmapValue* ret_bitmap) {
```
Comment (ucasfl) [inline be/src/util/bitmap_value.h:1698]: It's prefer `pass by value` then `pass by const reference` for basic data type.

File: be/src/util/bitmap_value.h [hunk 2]
```diff
@@ -1710,18 +1709,17 @@ class BitmapValue {
...
-                }
-                if (*it < range_end) {
-                    ret_bitmap->add(*it);
-                    ++count;
-                } else {
-                    break;
-                }
+            for (; it != _bitmap.end() && *it < range_end; ++it) {
+                ret_bitmap->add(*it);
+                ++count;
```
Comment (ucasfl) [inline be/src/util/bitmap_value.h:1720]: Reducing branch statement for better vectorization.

File: be/src/util/bitmap_value.h [hunk 3]
```diff
@@ -1710,18 +1709,17 @@ class BitmapValue {
             }
         }
         case BITMAP: {
+            auto it = _bitmap.begin();
+            for (; it != _bitmap.end() && *it < range_start;) {
```
Comment (cambyzju) [inline be/src/util/bitmap_value.h:1713]:
```suggestion
            while (it != _bitmap.end() && *it < range_start) {
```

**Comment by morningman** [comment]
Please describe your change

**Comment by ucasfl** [comment]
> Please describe your change

done.

**Pull request closed without merging**